packages/react-router/modules/useRouter.js

```diff
@@ -0,0 +1,20 @@
+import React, { useContext } from "react";
+import invariant from "tiny-invariant";
+
+import RouterContext from "./RouterContext";
+import matchPath from "./matchPath";
+
+export default function useRouter(options = {}) {
+  invariant(
+    typeof useContext === "function",
+    "The useRouter hook requires React 16.7 or greater"
+  );
+
+  let context = useContext(RouterContext);
+  let location = options.location || context.location;
```

Comment:
Maybe destructure instead? More concise.
```suggestion
  let { location } = options || context;
```

Reply:
That won't be the same result. It would use `options.location` as long as `options` object is defined.

Reply:
I don't think that works the same way

Reply:
Yea, it would always use `options.location` because `options` object is always defined.

Reply:
:doh: of course

Reply:
`const { location = context.location } = options` would the the alternative with destructuring